crossfire/__init__.py

```diff
@@ -57,20 +57,24 @@ def occurrences(
     )
 
 
+def _make_flatten(data, keys):
+    for item in data:
+        for key in keys:
+            item.update({f"{key}_{k}": v for k, v in item.get(key).items()})
+            item.pop(key)
+    return data
+
+
 def flatten(data, nested_columns=None):
-    origin_type = type(data)
     nested_columns = set(nested_columns or NESTED_COLUMNS)
     if not nested_columns.issubset(NESTED_COLUMNS):
         raise NestedColumnError(nested_columns)
-    if isinstance(data, DataFrame):
+    if isinstance(data, DataFrame) and not data.empty:
+        keys = set(data.columns) & nested_columns
         data = data.to_dict(orient="records")
+        return DataFrame(_make_flatten(data, keys))
```

Comment:
> converting `DAtaFrame` to `dict`

I still think this [is not a good strategy](https://github.com/FelipeSBarros/crossfire/pull/103#discussion_r1501812069), and you have not tried other alternatives. Let's focus on an alternative to it instead of advancing with this (arguably) unneeded transformation.